reactioncommerce/reaction-component-library
# fix: update packages to fix issues with build

Resolves https://github.com/reactioncommerce/reaction/issues/5460 (maybe?)
Impact: **critical**  
Type: **bugfix**

## Component

- Update various babel packages
- Update husky and commitlint because I couldn't commit. (the hook failed and prevented committing)

## Breaking changes

- Updated babel versions should mean that other projects that use this library should update their various babel packages to the latest versions. For example below

**devDependencies**
```
    "@babel/cli": "~7.5.5",
    "@babel/core": "~7.5.5",
    "@babel/node": "~7.5.5",
    "@babel/plugin-proposal-class-properties": "~7.5.5",
    "@babel/plugin-proposal-decorators": "~7.4.4",
    "@babel/plugin-proposal-export-namespace-from": "~7.2.0",
    "@babel/plugin-proposal-function-sent": "~7.2.0",
    "@babel/plugin-proposal-json-strings": "~7.2.0",
    "@babel/plugin-proposal-numeric-separator": "~7.2.0",
    "@babel/plugin-proposal-throw-expressions": "~7.2.0",
    "@babel/plugin-syntax-dynamic-import": "~7.2.0",
    "@babel/plugin-syntax-import-meta": "~7.2.0",
    "@babel/plugin-transform-runtime": "~7.5.5",
    "@babel/preset-env": "~7.5.5",
    "@babel/preset-react": "~7.0.0",
```

## Testing
1. Build should pass
2. Docs should render and all components should work. Most specifically the `Select` component and that's the single component causing all the issues...
3. `npm link` or otherwise link the built package to a project and verify that the app still runs and places where the `Select` component is used, are functioning correctly.
  - Checkout address form on example-storefront
  - Tags admin in the Reaction admin
